sfputil/main.py

```diff
@@ -678,148 +681,281 @@ def eeprom(port, dump_dom, namespace):
+        print_all_valid_port_values()
+        return ERROR_INVALID_PORT, f'Error: invalid port {logical_port_name}'
 
     if is_port_type_rj45(logical_port_name):
-        click.echo("{}: SFP EEPROM Hexdump is not applicable for RJ45 port".format(port))
-        sys.exit(ERROR_INVALID_PORT)
+        return ERROR_INVALID_PORT, f'{logical_port_name}: SFP EEPROM Hexdump is not applicable for RJ45 port'
 
+    physical_port = logical_port_to_physical_port_index(logical_port_name)
     try:
-        presence = platform_chassis.get_sfp(physical_port).get_presence()
+        sfp = platform_chassis.get_sfp(physical_port)
+        presence = sfp.get_presence()
     except NotImplementedError:
-        click.echo("Sfp.get_presence() is currently not implemented for this platform")
-        sys.exit(ERROR_NOT_IMPLEMENTED)
+        return ERROR_NOT_IMPLEMENTED, 'Sfp.get_presence() is currently not implemented for this platform'
 
     if not presence:
-        click.echo("SFP EEPROM not detected")
-        sys.exit(ERROR_NOT_IMPLEMENTED)
-    else:
-        try:
-            id = platform_chassis.get_sfp(physical_port).read_eeprom(0, 1)
-            if id is None:
-                click.echo("Error: Failed to read EEPROM for offset 0!")
-                sys.exit(ERROR_NOT_IMPLEMENTED)
-        except NotImplementedError:
-            click.echo("Sfp.read_eeprom() is currently not implemented for this platform")
-            sys.exit(ERROR_NOT_IMPLEMENTED)
+        return ERROR_NOT_IMPLEMENTED, 'SFP EEPROM not detected'
 
-        if id[0] == 0x3:
-            output = eeprom_hexdump_sff8472(port, physical_port, page)
+    try:
+        api = sfp.get_xcvr_api()
+        if not api:
+            return ERROR_NOT_IMPLEMENTED, 'Error: Failed to read EEPROM for offset 0!'
+
+        from sonic_platform_base.sonic_xcvr.api.public import sff8636, sff8436, cmis, sff8472
+        from sonic_platform_base.sonic_xcvr.fields import consts
+        if isinstance(api, cmis.CmisApi):
+            if page is None: # print all possible pages
+                if api.is_flat_memory():
+                    pages = [0]
+                else:
+                    pages = [0, 1, 2, 16, 17]
+                    if api.is_coherent_module():
+                        pages.extend([0x30, 0x31, 0x32, 0x33, 0x34, 0x35, 0x38, 0x39, 0x3a, 0x3b])
+                    cdb_support = api.xcvr_eeprom.read(consts.CDB_SUPPORT)
+                    if cdb_support != 0:
+                        pages.append(0x9f)
+            else:
+                pages = [0]
+                if page not in pages:
+                    pages.append(page)
+            return eeprom_hexdump_pages_general(logical_port_name, pages, page)
+        elif isinstance(api, sff8636.Sff8636Api) or isinstance(api, sff8436.Sff8436Api):
+            if page is None:
+                if api.is_flat_memory():
+                    pages = [0]
+                else:
+                    pages = [0, 1, 2, 3]
```

Comment:
@Junchao-Mellanox can we define this list of pages as SFF8436_MODULE_PAGE = [] at the top?